plugins/misc/w_helicity.h

```diff
@@ -0,0 +1,186 @@
+// -*- C++ -*-
+// author: afiq anuar
+// short: for computing ttbar spin correlation variables
+
+#ifndef FWK_W_HELICITY_H
+#define FWK_W_HELICITY_H
+
+#include "misc/container_util.h"
+#include "misc/constants.h"
+#include "misc/numeric_vector.h"
+#include <cmath>
+
+template <typename Number = float>
+const std::vector<std::pair<std::string, Number>>&
+compute_w_helicity(Number pTop_pt, Number pTop_eta, Number pTop_phi, Number pTop_m,
+                         Number aTop_pt, Number aTop_eta, Number aTop_phi, Number aTop_m,
+                         Number pLep_pt, Number pLep_eta, Number pLep_phi, Number pLep_m,
+                         Number aLep_pt, Number aLep_eta, Number aLep_phi, Number aLep_m,
+                         Number pW_pt, Number pW_eta, Number pW_phi, Number pW_m,
+                         Number aW_pt, Number aW_eta, Number aW_phi, Number aW_m,
+                         Number pB_pt, Number pB_eta, Number pB_phi, Number pB_m,
+                         Number aB_pt, Number aB_eta, Number aB_phi, Number aB_m)
+
+{
+  using namespace Framework;
+  static std::vector<std::pair<std::string, Number>> m_w_hel;
+  static int initialize = 0;
```

Comment:
consider using thread_local instead of static, as eventually the fwk will go multithreading.